tests/functional/test_linux_appimage.py

```diff
@@ -0,0 +1,58 @@
+#-----------------------------------------------------------------------------
+# Copyright (c) 2005-2020, PyInstaller Development Team.
+#
+# Distributed under the terms of the GNU General Public License (version 2
+# or later) with exception for distributing the bootloader.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#
+# SPDX-License-Identifier: (GPL-2.0-or-later WITH Bootloader-exception)
+#-----------------------------------------------------------------------------
+
+"""
+GNU/Linux-specific test to check the bootloader from the AppImage.
+"""
+
+# Library imports
+# ---------------
+import os
+import pathlib
+import stat
+import subprocess
+import pytest
+
+
+@pytest.mark.linux
+@pytest.mark.parametrize('arch', ['x86_64'])
+def test_appimage_loading(tmp_path, pyi_builder_spec, arch):
+    # Skip the test if appimagetool is not found
+    appimagetool = pathlib.Path.home() / ('appimagetool-%s.AppImage' % arch)
+    if not os.path.isfile(appimagetool):
+        pytest.skip('%s not found' % appimagetool)
+
+    # Ensure appimagetool is executable
+    if not os.access(appimagetool, os.X_OK):
+        os.chmod(appimagetool, stat.S_IXUSR)
```

Comment:
This `os.chmod()` is destructive, in the sense that it will reset all other permission bits.